go/vt/vtgate/planbuilder/merge.go

```diff
@@ -0,0 +1,129 @@
+	}, nil
+}
+
+// Symtab returns the associated symtab.
+func (mb *merge) Symtab() *symtab {
+	panic("unreachable")
+	// return mb.symtab
+}
+
+// SetSymtab sets the symtab for the current node and
+// its non-subquery children.
+func (mb *merge) SetSymtab(*symtab) {
+	panic("unreachable")
+	// mb.symtab = symtab
+	// mb.Left.SetSymtab(symtab)
+	// mb.Right.SetSymtab(symtab)
+}
+
+// Order is a number that signifies execution order.
+// A lower Order number Route is executed before a
+// higher one. For a node that contains other nodes,
+// the Order represents the highest order of the leaf
+// nodes. This function is used to travel from a root
+// node to a target node.
+func (mb *merge) Order() int {
+	panic("unreachable")
+}
+
+// SetOrder sets the order for the underlying routes.
+func (mb *merge) SetOrder(int) {
+	panic("unreachable")
+}
+
+// Primitve returns the underlying primitive.
+func (mb *merge) Primitive() engine.Primitive {
+	return mb.emerge
+}
+
+// Leftmost returns the leftmost route.
+func (mb *merge) Leftmost() *route {
+	panic("unreachable")
+}
+
+// Join joins the two builder objects. The outcome
+// can be a new builder or a modified one.
+func (mb *merge) Join(rhs builder, ajoin *sqlparser.JoinTableExpr) (builder, error) {
+	return newJoin(mb, rhs, ajoin)
+}
+
+// SetRHS marks all routes under this node as RHS due
+// to a left join. Such nodes have restrictions on what
+// can be pushed into them. This should not propagate
+// to subqueries.
+func (mb *merge) SetRHS() {
+	panic("unreachable")
+}
+
+// PushSelect pushes the select expression through the tree
+// all the way to the route that colsym points to.
+// PushSelect is similar to SupplyCol except that it always
+// adds a new column, whereas SupplyCol can reuse an existing
+// column. The function must return a colsym for the expression
+// and the column number of the result.
+func (mb *merge) PushSelect(expr *sqlparser.NonStarExpr, rb *route) (colsym *colsym, colnum int, err error) {
+	panic("unreachable")
+}
+
+// PushOrderByNull pushes the special case ORDER By NULL to
+// all routes. It's safe to push down this clause because it's
+// just on optimization hint.
+func (mb *merge) PushOrderByNull() error {
+	panic("unreachable")
+}
+
+// PushMisc pushes miscelleaneous constructs to all the routes.
+// This should not propagate to subqueries.
+func (mb *merge) PushMisc(sel *sqlparser.Select) error {
+	panic("unreachable")
+}
+
+// Wireup performs the wire-up work. Nodes should be traversed
+// from right to left because the rhs nodes can request vars from
+// the lhs nodes.
+func (mb *merge) Wireup(bldr builder, jt *jointab) error {
+	err := mb.Right.Wireup(bldr, jt)
+	if err != nil {
+		return err
+	}
+	return mb.Left.Wireup(bldr, jt)
+}
+
+// SupplyVar finds the common root between from and to. If it's
+// the common root, it supplies the requested var to the rhs tree.
+func (mb *merge) SupplyVar(from, to int, col *sqlparser.ColName, varname string) error {
```

Comment:
You may have to build this functionality because we're allowing unions in subqueries. Let me know if you need help here. Same for SupplyCol.